lib/puppet/pops/functions/function.rb

```diff
@@ -89,7 +89,11 @@ def internal_call_function(scope, function_name, args, &block)
     raise ArgumentError, "Function #{self.class.name}(): cannot call function '#{function_name}' - no loader specified" unless the_loader
 
     func = the_loader.load(:function, function_name)
-    return func.call(scope, *args, &block) if func
+    if func
+      Puppet::Util::Profiler.profile(function_name, [:functions, function_name]) do
```

Comment:
Is this the same way we generate the metric IDs for the old functions?  I ask because in Puppet Server we surface metrics about function calls in a few places such as our status endpoint.  As long as this naming convention matches what we were doing for the old functions then these should Just Work.